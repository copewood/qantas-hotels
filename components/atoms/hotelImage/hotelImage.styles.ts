import styled from 'styled-components';

export const StyledImageContainer = styled.div`
  width: 150px;
  margin-right: 2rem;
  margin-bottom: 1rem;
  position: relative;
`;

export const StyledImage = styled.img`
  width: 150px;
`;

export const StyledText = styled.div`
  position: absolute;
  background-color: #fff;
  top: 10px;
  left: 10px;
`;