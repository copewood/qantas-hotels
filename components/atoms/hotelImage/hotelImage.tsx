import { Offer, PreviewImage } from '@components/models/hotel';
import React, { FC } from 'react';
import { 
  StyledImageContainer,
  StyledImage,
  StyledText
} from './hotelImage.styles';

type HotelImageProps = {
   previewImage: PreviewImage;
   offer: Offer;
};

export const HotelImage: FC<HotelImageProps> = ({previewImage, offer}) => {
  return <>
    <StyledImageContainer>
      <StyledImage src={previewImage.url} />
      <StyledText>{offer && offer.promotion.title}</StyledText>
    </StyledImageContainer>
  </>;
};
