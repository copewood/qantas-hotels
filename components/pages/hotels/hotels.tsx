import React, { FC, useEffect, useState} from 'react';

import { HotelListHeader } from 'components/features/hotelListHeader/hotelListHeader';
import { HotelList } from 'components/features/hotelList/hotelList';
import { Hotel } from '@components/models/hotel';

type HotelsProps = {};

const fetchHotels = async () => {
    return fetch('/api/hotels')
        .then((response) => response.json())
        .then((responseJson) => responseJson);
}

export const Hotels: FC<HotelsProps> = ({}) => {
  const [hotels, setHotels] = useState<Array<Hotel>>();

  useEffect(() => {
    fetchHotels().then((response)=>{
      setHotels(response.results as Array<Hotel>)
    });
  },[])
  
  return <>
    {hotels && <>
      <HotelListHeader hotels={hotels}/>
      {/* <HotelList hotels={hotels}/> */}
    </>}
  </>;
};
