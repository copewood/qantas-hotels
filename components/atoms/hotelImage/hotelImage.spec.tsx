import React from 'react';
import { render } from '@testing-library/react';
import { HotelImage } from './hotelImage';
import { mockHotels } from '../../../pages/api/hotels'

describe('HotelList', () => {

  it('should show HotelImage', () => {
    const { container } = render(<HotelImage previewImage={mockHotels.results[0].property.previewImage}/>);
    expect(container).toMatchSnapshot()
  });
});
