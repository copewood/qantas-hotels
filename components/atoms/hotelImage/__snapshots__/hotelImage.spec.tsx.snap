// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HotelList should show HotelImage 1`] = `
<div>
  <div
    class="sc-bczRLJ bxsSWL"
  >
    <img
      class="sc-gsnTZi bFjRRX"
      src="https://unsplash.it/145/125/?random"
    />
    <div
      class="sc-dkzDqf iuODgQ"
    />
  </div>
</div>
`;
